### ISSUE TITLE
[1.12] problems with greenhouse

### ISSUE THREAD

**Issue by nouille07** [ForestryMC/ForestryMC#1849]
So a few problems, I'm playing in 1.12 with forestry version 5.5.5.192. I don't really know if the greehouse is working or still work in progress but when I try to mess around in creative I get a crash when puttin a heater (or any other regulator block) on top of the fomed multiblock base (5*5*2 of greenhouse block with a gearbox somewhere in the bottom), which is a bit disapointing as I started rafting the blocks in my current world and wanted to use bees.

I'm also not sure to understand how it works, does the biome changes (assuming it's working) only happens above the greenhouse multiblock? what are the windows supposed to do? some pictures for how to set it up could be very useful before having a fully detailed wiki entry.

Nedelosk help me! I saw in another issue that you are looking for testers? I'd be happy to help if that could accelerate me playing with bees again! :D 

here's the crash log link the crash happens in my modpack and with just forestry (dunno if I did it right, new to github): https://gist.github.com/nouille07/ea04e192c6a57d667a4d625e4a0521ea

**Comment by nouille07** [comment]
ok the crash is gone with the last update but I still can't figure out how it works, where do I put the blocks like the heater?

**Comment by Nedelosk** [comment]
# Greenhouse
First you have to build the multiblock. The multiblock can have a size from 3x2x3 (18 Blocks) to 5x5x5 (125 Blocks)
### 3x1x3
![2017-09-18_13 58 02](https://user-images.githubusercontent.com/10259214/30541177-8c86036c-9c7a-11e7-8630-5a7a4d2f0286.png)
### 3x2x3
![2017-09-18_13 58 10](https://user-images.githubusercontent.com/10259214/30541179-8c8da2d4-9c7a-11e7-8f80-fac180321f30.png)
### 5x2x5
![2017-09-18_13 58 30](https://user-images.githubusercontent.com/10259214/30541178-8c875618-9c7a-11e7-93e8-6128bf1b7669.png)
### 5x5x5
![2017-09-18_13 59 11](https://user-images.githubusercontent.com/10259214/30541180-8c91852a-9c7a-11e7-8811-eb02e886cfc5.png)
Than you have to build a platform.
![2017-09-18_14 07 21](https://user-images.githubusercontent.com/10259214/30541220-b688bccc-9c7a-11e7-964e-36b103de53ec.png)
Ther are only only two conditions that have to be fulfilled.
The first condition is that the greenhouse has to be closed on the x and z axis on a specifiy height. And the distance from the wall to the center of the greenhouse must not be biger than `the
 size of the greenhouse on this axis (3 - 5) multiplied with the "greenhouseSize"config value (default: 4)`.
![2017-09-18_14 34 30](https://user-images.githubusercontent.com/10259214/30542157-80917ac4-9c7e-11e7-909a-29e67b2f529e.png)
You can see height if you activate the preview mode on the greenhouse screen item. And you can change it with a shift right click if you have the item in your hand.
![2017-09-18_14 30 34](https://user-images.githubusercontent.com/10259214/30542012-fdbfd33e-9c7d-11e7-9d47-a1a772ff7e34.png)
And the second condition is that you have to set a block above every block that is within the closed area of the first condition.
![2017-09-18_14 29 29](https://user-images.githubusercontent.com/10259214/30542114-57c04cce-9c7e-11e7-98f5-bae3b09be2d6.png)
Technickly you could build a structure like on the picture above. But thats not very beautiful and every wall block and every block inside the structure can be used to change the climate of the greenhouse if you set climate sources on this positions. 
![2017-09-18_14 05 59](https://user-images.githubusercontent.com/10259214/30541183-8e976c2c-9c7a-11e7-998e-e0fb745c5087.png)
## Climate Sources
Heaters, fans, humidifiers, dehumidifiers are using energy to change the climate. The window trys to change the climate of the greenhouse to the climate of the biome and the hygroregulator used lava or water to change the climate.
![2017-09-18_14 51 38](https://user-images.githubusercontent.com/10259214/30542762-e32b473a-9c80-11e7-9525-c8733ca9d97d.png)
A greenhouse with a big structure on its top needs more climate sources or climate sources with a circuit board than a greenhouse with a small structure on its top.

Sorry for my bad english

**Comment by nouille07** [comment]
I get it now! I'll try it as soon as I can! bee power is mine now!!

thanks a lot ned :)

**Comment by KorDum** [comment]
I tested greenhouse in forestry 5.7.0.238. But all climate sources show me error message, no matter what the size of the multiblock (I built 3x2x3 and 5x5x5).

![image](https://user-images.githubusercontent.com/1331559/35643019-0479bbd2-06d6-11e8-86ac-82910ea09eec.png)

![image](https://user-images.githubusercontent.com/1331559/35643026-0c6a8c72-06d6-11e8-8da9-bb4437260c19.png)

Did I do something wrong?
